docs/batch_size_optimizer/server.md

```diff
@@ -0,0 +1,156 @@
+
+## Should I use this?
+
+The key of BSO is recurrent training. If you are training your model periodically or repeatedly, BSO can be a great choice to reduce energy or time consumption.
+
+## Limitations
+
+We currently doesn't support heterogeneous GPUs or different configurations. Number of GPUs, gpu models, and other configurations in JobSpec should be identical in recurrent trainings. If you are running your training in a various environment each time, then it might not desirable to use BSO.
+
+## Quick start (Server)
+
+1. Clone the repository
+
+    ```Shell
+    git clone https://github.com/ml-energy/zeus/tree/master
+    ```
+
+2. Create `.env` under `/docker`. Example of `.env` is provided below.
+
+    For default, we are using the MySQL for the database.
+
+    ```Shell
+    ZEUS_BSO_DB_USER="me" 
+    ZEUS_BSO_DB_PASSWORD="secret"
+    ZEUS_BSO_ROOT_PASSWORD="secret*"
+    ZEUS_BSO_SERVER_PORT=8000
+    ZEUS_BSO_LOG_LEVEL = "INFO"
+    ZEUS_BSO_ECHO_SQL = "True"
+    ```
+
+    If you want to use different databases, you need to add `ZEUS_BSO_DATABASE_URL` as an environment variable. See [Remark](#remark-about-server) for detail.
+    Also, if you are running using docker-compose or kubernetes, you need to change the image name under `db` in docker-compose file.
+
+3. Running a server
+    - Using docker-compose
+
+        ```Shell
+        cd docker 
+        docker-compose -f ./docker/docker-compose.yaml up
+        ```
+
+        This will build images for each container: db, migration and the server. Then, it will spin those containers.
+
+    - Using Kubernetes.
+        1. Build an image.
+
+            ```Shell
+            # From the root directory
+            build -f ./docker/server.Dockerfile -t bso-server . 
+            build -f ./docker/migration.Dockerfile -t bso-server .
+            ```
+
+        2. Create kubernetes yaml files using Kompose.
+
+            ```Shell
+            cd docker 
+            docker-compose config > docker-compose-resolved.yaml && kompose convert -f docker-compose-resolved.yaml -o ./kube/ && rm docker-compose-resolved.yaml
+            ```
+
+            It first resolves env files using docker-compose, then create kubernetes yaml files under `docker/kube/`
+
+        3. Run kubernetes.
+
+            ```Shell
+            cd kube
+            kubectl apply -f .
+            ```
+
+    - Using uvicorn.
+
+        If you are using the uvicorn to spin the server, you need to create database and perform migration before starting the server.
+
+        1. Run the database of your choice.
+        2. Set the environment variables in `.env`
+
+            ```Shell
+            ZEUS_BSO_DATABASE_URL="me" 
+            ZEUS_BSO_LOG_LEVEL = "INFO"
+            ZEUS_BSO_ECHO_SQL = "True"
+            ```
+
+        3. Run Alembic migration by following the guides in `zeus/optimizer/batch_size/migrations/README.md`
```

Comment:
Source code and usage guides are to be separated. Move this README into actual docs.